Source/Core/Core/AchievementManager.cpp

```diff
@@ -682,6 +685,26 @@ void AchievementManager::HandleAchievementTriggeredEvent(const rc_runtime_event_
                                 Config::Get(Config::RA_ENCORE_ENABLED));
 }
 
+void AchievementManager::HandleAchievementProgressUpdatedEvent(
+    const rc_runtime_event_t* runtime_event)
+{
+  if (!Config::Get(Config::RA_PROGRESS_ENABLED))
+    return;
+  auto it = m_unlock_map.find(runtime_event->id);
+  if (it == m_unlock_map.end())
+    return;
+  AchievementId game_data_index = it->second.game_data_index;
+  FormattedValue value{};
+  if (rc_runtime_format_achievement_measured(&m_runtime, runtime_event->id, value.data(),
```

Comment:
Surprised we use '0' here.  Is there an API return value we can use instead?

Reply:
Should we handle this error somehow?

Reply:
It's returning the result of the sprintf inside it, which is the length of the string inside the buffer.

Does Dolphin have an error log I should be using when things like this happen?

Reply:
Someone more familiar with these things would have to confirm but I'd think we would want a log category specific for achievements.  I guess it might depend on how many log messages you have...

Reply:
I'm asking more generically than this - I have no logging in the achievements thus far because I don't know how Dolphin logs things.

Reply:
Oh, sorry!  For example, to log an error you'd use:  https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/Logging/Log.h#L119

Usage:

```
ERROR_LOG_FMT(
            MEMCARD_MANAGER,
            "A fabulous error for memcard manager by user '{}'",
            "iwubcode");
```